app/services/RestaurantService.java

```diff
@@ -176,82 +227,122 @@ public Restaurant getRestaurantWithId(final UUID id) {
-			restaurantReview = new RestaurantReview(
-					reviewForm.getRestaurantId(),
-					user.getId(),
-					reviewForm.getReviewScore(),
-					reviewForm.getReviewText()
-			);
-		} else {
-			restaurantReview.setReview(reviewForm.getReviewText());
-			restaurantReview.setRating(reviewForm.getReviewScore());
-		}
-
-		getSession().save(restaurantReview);
-		return true;
+	getSession().save(restaurantReview);
+	return true;
+    }
+
+    /**
+     * Gets number of restaurants.
+     *
+     * @return the number of restaurants
+     */
+    public Long getNumberOfRestaurants()
+    {
+	return Long.valueOf(getSession().createCriteria(Restaurant.class).setProjection(Projections.rowCount())
+		.uniqueResult().toString());
+    }
+
+    /**
+     * Update picture string.
+     *
+     * @param imageUploadForm
+     *            the image upload form
+     * @return the string
+     * @throws Exception
+     *             the exception
+     */
+    public String updatePicture(final ImageUploadForm imageUploadForm) throws Exception
+    {
+	Restaurant restaurant = (Restaurant) getSession().createCriteria(Restaurant.class)
+		.add(Restrictions.eq("id", imageUploadForm.getRestaurantId())).uniqueResult();
+
+	String newImagePath = AWS_BASE_PATH + imageUploadForm.getRestaurantId() + "-" + imageUploadForm.getImageType()
+		+ "." + imageUploadForm.getExtension();
+
+	if (imageUploadForm.getImageType().equals("profile"))
+	{
+	    restaurant.setProfileImagePath(newImagePath);
+	} else
+	{
+	    restaurant.setCoverImagePath(newImagePath);
 	}
 
-	/**
-	 * Gets number of restaurants.
-	 *
-	 * @return the number of restaurants
-	 */
-	public Long getNumberOfRestaurants() {
-		return Long.valueOf(getSession().createCriteria(Restaurant.class)
-				.setProjection(Projections.rowCount())
-				.uniqueResult().toString());
-	}
+	getSession().update(restaurant);
+	return "{ \"imageFor\": \"" + imageUploadForm.getImageType() + "\", \"url\": \"" + newImagePath + "\"}";
+    }
 
-	/**
-	 * Update picture string.
-	 *
-	 * @param imageUploadForm the image upload form
-	 * @return the string
-	 * @throws Exception the exception
-	 */
-	public String updatePicture(final ImageUploadForm imageUploadForm) throws Exception {
-		Restaurant restaurant = (Restaurant) getSession().createCriteria(Restaurant.class)
-				.add(Restrictions.eq("id", imageUploadForm.getRestaurantId()))
-				.uniqueResult();
-
-		String newImagePath = AWS_BASE_PATH + imageUploadForm.getRestaurantId() + "-" + imageUploadForm.getImageType() + "." + imageUploadForm.getExtension();
-
-		if (imageUploadForm.getImageType().equals("profile")) {
-			restaurant.setProfileImagePath(newImagePath);
-		} else {
-			restaurant.setCoverImagePath(newImagePath);
-		}
+    @SuppressWarnings("unchecked")
+    private UserHabitsBean getUserHabits(UUID userId)
+    {
+
+	Integer price = 0;
+
+	UUID cityID = (UUID) getSession().createCriteria(User.class).setProjection(Projections.property("city.id"))
+		.uniqueResult();
```

Comment:
So, we base our recommendations based on the users home location? What if they open the app on their phone/laptop while traveling? Maybe the actual device location should be used instead?